android/build.sh

```diff
@@ -20,8 +20,35 @@ THIS_SCRIPT="$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BA
 #
 SHLVL=0
 
+clean ( ) {
+  if [ -f "$KEYMAN_ROOT/android/KMAPro/kMAPro/libs/keyman-engine.aar" ]; then
+    echo "Cleaning keyman-engine.aar"
+    rm -rf "$KEYMAN_ROOT/android/KMAPro/kMAPro/libs/keyman-engine.aar"
+  fi
+  if [ -d "$KEYMAN_ROOT/android/KMAPro/kMAPro/build/outputs" ]; then
+    echo "Cleaning KMAPro build outputs"
+    rm -rf "$KEYMAN_ROOT/android/KMAPro/kMAPro/build/outputs"
+  fi
+  if [ -d "$KEYMAN_ROOT/android/upload" ]; then
+    echo "Cleaning upload directory"
+    rm -rf "$KEYMAN_ROOT/android/upload"
+  fi
+}
+
 echo Build KMEA and KMAPro:
 
+# Parse args
+while [[ $# -gt 0 ]] ; do
+    key="$1"
+    case $key in
+        -clean)
+            clean
+            ;;
+    esac
+    shift # past argument
+done
```

Comment:
Won't this lose all our args that we pass with `$@` later?

Reply:
argh, you're right.

Maybe we don't bother with the flag `-clean` and always do `clean`?

Reply:
You could just test `$1` and `shift`:

```sh
if [[ "$1" == "-clean" ]]; then
  shift
  clean
fi
```